### ISSUE TITLE
reduce_max returns wrong shape

### ISSUE THREAD

**Issue by comeonfox** [tensorflow/tensorflow#23852]
**Problem**
**tf.reduce_max returns tensor with wrong shape**

    ...
    left_expr = tf.reduce_sum(left_encs, axis=1)
    left_expr = tf.Print(left_expr,  [tf.shape(left_encs), tf.shape(left_expr)], message="Shapes Are: ")
    ...

and the output is:

    Shapes Are: [46 21 512][1 512]

the tensor `left_encs` is the output of transformer encoder, the meaning of the shape is [batch_size, sequence_length, hidden_dim]

I have tried to construct a random tensor with same shape as `left_encs` and the call `tf.reduce_sum(left_encs, axis=1)` and the output is as expected `[46, 512]`.

but I cannot get it right from the real code.

is there any operation should I do before calling `tf.reduce_sum`?

**System information**
- TensorFlow installed from: Anaconda
- TensorFlow version: v1.11
- Python version: 3.6
- CUDA/cuDNN version: v9.0
- GPU model and memory: Tesla K40

**Comment by ymodak** [comment]
In order to expedite the trouble-shooting process, please provide a code snippet to reproduce the issue reported here. Thanks!

**Comment by ymodak** [comment]
Closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
